File: Vehicle-Configurator-.NET/backed-.NET/Models/Vehicle.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace backed_.NET.Models;


public enum CompType
{
    C,
    E,
    I,
    S
}

public enum IsConfigurable
{
    N,
    Y
}

[Table("vehicle")]
[Index("ModelId", Name = "FK27cc9wpp4hrwtmjyew3ln1pw3")]
[Index("CompId", Name = "FKsa5e1l7iqlfcfeteqs7osy4ey")]
public partial class Vehicle
{
    [Key]
    [Column("confi_id")]
    public int ConfiId { get; set; }

    [Column("comp_type")]
    public CompType CompType { get; set; }

    [Column("is_configrable")]
    public IsConfigurable IsConfigrable { get; set; }

    [Column("comp_id")]
    public int CompId { get; set; }

    [Column("model_id")]
    public int? ModelId { get; set; }

    [ForeignKey("CompId")]
    [InverseProperty("Vehicles")]
    public virtual Component Comp { get; set; } = null!;

    [ForeignKey("ModelId")]
    [InverseProperty("Vehicles")]
    public virtual Model? Model { get; set; }
}

File: Vehicle-Configurator-Java/backend/src/main/java/com/example/entities/Vehicle.java
package com.example.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

enum Comp_Type {
	C, S, I, E
}

enum Is_Configurable {
	N, Y
}

@Entity
@Table(name = "Vehicle")
public class Vehicle {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int confi_id;

	@ManyToOne
	@JoinColumn(name = "comp_id", nullable = false)
	private Component component;
	
	@ManyToOne
	@JoinColumn(name = "model_id")
	private Model model ;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	private Comp_Type comp_type;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	private Is_Configurable is_configrable;

	public int getConfi_id() {
		return confi_id;
	}

	public void setConfi_id(int confi_id) {
		this.confi_id = confi_id;
	}

	public Component getComponent() {
		return component;
	}

	public void setComponent(Component component) {
		this.component = component;
	}

	public Model getModel() {
		return model;
	}

	public void setModel(Model model) {
		this.model = model;
	}

	public Comp_Type getComp_type() {
		return comp_type;
	}

	public void setComp_type(Comp_Type comp_type) {
		this.comp_type = comp_type;
	}

	public Is_Configurable getIs_configrable() {
		return is_configrable;
	}

	public void setIs_configrable(Is_Configurable is_configrable) {
		this.is_configrable = is_configrable;
	}

	@Override
	public String toString() {
		return "Vehicle [confi_id=" + confi_id + ", component=" + component + ", model=" + model + ", comp_type="
				+ comp_type + ", is_configrable=" + is_configrable + "]";
	}
	
	


}